syntax = "proto3";

package BookStore;

service Books {
  rpc GetBook(GetBookRequest) returns (GetBookResponse) {}
  rpc GetBookById(GetBookByIdRequest) returns (GetBookByIdResponse) {}
  rpc AddBook(AddBookRequest) returns (AddBookResponse) {}
  rpc UpdateBook(UpdateBookRequest) returns (UpdateBookResponse) {}
  rpc DeleteBook(DeleteBookRequest) returns (DeleteBookResponse) {}
}

message Book {
  optional string id = 1;
  optional string title = 2;
  optional string author = 3;
  optional string description = 4;
  optional string price = 5;
  optional string publisher = 6;
  optional string published_date = 7;
  optional string updated_date = 8;
}

message GetBookRequest {}

message  GetBookResponse {
  repeated Book books = 1;
}

message GetBookByIdRequest {
  optional string id = 1;
}

message GetBookByIdResponse {
  Book book = 1;
}

message AddBookRequest {
  optional Book book = 1;
}

message AddBookResponse {
  optional string id = 1;
}

message UpdateBookRequest {
  optional Book book = 1;
}

message UpdateBookResponse {
  optional string id = 1;
}

message DeleteBookRequest {
  optional string id = 1;
}

message DeleteBookResponse {
  optional string id = 1;
}

